subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -360,6 +360,30 @@ static struct bass_recv_state_internal *get_free_recv_state(void)
 		}
 	}
 
+	LOG_DBG("No inactive receive state available");
+
+#if defined(CONFIG_SD_RECEIVE_STATE_OVERWRITE_OLDEST)
+	/* Free a receive state that is not currently being synced,
+	 * starting from the oldest in terms of when it was updated
+	 */
+	for (size_t i = 0U; i < ARRAY_SIZE(scan_delegator.recv_states); i++) {
+		struct bass_recv_state_internal *free_internal_state =
+					&scan_delegator.recv_states[i];
+
+		if (free_internal_state->state.pa_sync_state == BT_BAP_PA_STATE_NOT_SYNCED ||
+			free_internal_state->state.pa_sync_state == BT_BAP_PA_STATE_FAILED ||
+			free_internal_state->state.pa_sync_state == BT_BAP_PA_STATE_INFO_REQ) {
```

Comment:
This should also take the BIG sync state into account, as you can be BIG synced without a PA sync